driver: watchdog: Error handling issues in wdt_wwdg_stm32.c

FRASTM opened the pull request:
This patch changes wwdg_stm32_get_pclk function to test the return value of clock_control_get_rate
It fixes the issue Error handling issues in /drivers/watchdog/wdt_wwdg_stm32.c #20503 
[Coverity CID :205655]

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango reviewed (changes requested):
I'm good with the change, but let's extend this to other drivers too.

FRASTM commented:
Completes the fix of Error handling issues in /drivers/watchdog/wdt_wwdg_stm32.c [#20503]
applying the test of retrun code clock_control_get_rate in other drivers:  i2c, pwm, uart, spi

FRASTM commented:
include logging/log.h

FRASTM committed:
driver: serial: Error handling issues in uart_stm32_set_baudrate

This patch tests the return code when calling clock_control_get_rate
and completes the issue #20503 seen on watchdog
[Coverity CID :205655]

Fixes #20503

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on drivers/watchdog/wdt_wwdg_stm32.c:
```diff
@@ -84,7 +90,7 @@ static u32_t wwdg_stm32_get_timeout(struct device *dev, u32_t prescaler,
 {
 	u32_t divider = WWDG_INTERNAL_DIVIDER * (1 << (prescaler >> 7));
-	float f_wwdg = (float)wwdg_stm32_get_pclk(dev) / divider;
+	float f_wwdg = wwdg_stm32_get_pclk(dev) / divider;
```
This change is not part of this issue.

FRASTM replied:
this is covered by the PullRequest 
driver: watchdog: Incorrect expression in wdt_wwdg_stm32.c [#20617]

erwango commented on drivers/spi/spi_ll_stm32.c:
```diff
@@ -281,9 +284,11 @@ static int spi_stm32_configure(struct device *dev,
+			(clock_control_subsys_t) &cfg->pclken, &clock) < 0) {
+		LOG_ERR("Failed call clock_control_get_rate");
+		return -EIO;
+	}
```
Break a line after `}`

FRASTM replied:
ok, done

erwango commented on drivers/pwm/pwm_stm32.c:
```diff
@@ -160,9 +163,11 @@ static int pwm_stm32_get_cycles_per_sec(struct device *dev, u32_t pwm,
+			(clock_control_subsys_t *)&cfg->pclken, &bus_clk) < 0) {
+		LOG_ERR("Failed call clock_control_get_rate");
+		return -EIO;
+	}
```
Break a line after `}`

FRASTM replied:
ok, done

erwango commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -14,7 +14,6 @@
 #include <drivers/i2c.h>
 #include "i2c_ll_stm32.h"
 
-#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
```
This would deserve a dedicated clean up commit.

FRASTM replied:
removed

>

erwango replied:
Still there as I can see

FRASTM replied:
lines 
`#define LOG_LEVEL_CONFIG_I2C_LOG_LEVEL`
`#include <logging/log.h>`
`LOG_MODULE_REGISTER(i2c_ll_stm32);`
do not changed

FRASTM committed:
driver: spi: Error handling issues in spi_stm32_configure

This patch tests the return code when calling clock_control_get_rate
and completes the issue  #20503 seen on watchdog
[Coverity CID :205655]

Fixes #20503

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
driver: pwm: Error handling issues in pwm_stm32_get_cycles_per_sec

This patch tests the return code when calling clock_control_get_rate
and completes the issue  #20503 seen on watchdog
[Coverity CID :205655]

Fixes #20503

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
driver: watchdog: Error handling issues in wdt_wwdg_stm32.c #20503

This patch tests the return code when calling clock_control_get_rate
[Coverity CID :205655]

Fixes #20503

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase

FRASTM commented:
rebase

FRASTM committed:
driver: i2c: Error handling issues in i2c_stm32_runtime_configure

This patch tests the return code when calling clock_control_get_rate
and completes the issue  #20503 seen on watchdog
[Coverity CID :205655]

Fixes #20503

Signed-off-by: Francois Ramu <francois.ramu@st.com>
